archivist/confirmer.py

```diff
@@ -94,20 +95,18 @@ def _wait_for_confirmation(self: Managers, identity: str) -> ReturnTypes:
             f"cannot confirm {identity} as confirmation_status is not present"
         )
 
-    if entity[CONFIRMATION_STATUS] == ConfirmationStatus.FAILED.name:
+    status = entity[CONFIRMATION_STATUS]
+    if status == ConfirmationStatus.FAILED.name:
         raise ArchivistUnconfirmedError(
             f"confirmation for {identity} FAILED - this is unusable"
         )
 
     # Simple hash
-    if entity[CONFIRMATION_STATUS] == ConfirmationStatus.CONFIRMED.name:
+    if status == ConfirmationStatus.CONFIRMED.name:
         return entity
 
     # merkle_log
-    if (
-        ConfirmationStatus[entity[CONFIRMATION_STATUS]].value
-        >= ConfirmationStatus.COMMITTED.value
-    ):
+    if ConfirmationStatus[status].value >= ConfirmationStatus.COMMITTED.value:
```

Comment:
Are you sure about this ? CONFIRMED is numerically lower than COIMMITTED

Reply:
I have modified comment

Reply:
code changed to make it clearer